test/Kestrel.Core.Tests/Http2ConnectionTests.cs

```diff
@@ -2190,6 +2212,88 @@ public Task HEADERS_Received_HeaderBlockContainsTEHeader_ValueIsNotTrailers_Stre
             Assert.Contains(5, _abortedStreamIds);
         }
 
+        [Fact]
+        public async Task GOAWAY_Received_RelievesConnectionBackpressure()
```

Comment:
While this is the current behavior, it's not the expected behavior. See https://github.com/aspnet/KestrelHttpServer/issues/2703. A connection can behave normally after a GoAway is received up until the connection is prematurely closed by the client. We may opt to differentiate by GoAway error code and terminate early for some of them.

Reply:
I think we just update this test when we change the GOAWAY behavior. I think this test might be fine since the client sends a GOAWAY frame with a Last-Stream-ID of 0 effectively aborting all three streams.